src/loop.c

```diff
@@ -95,7 +95,7 @@ static void luv_walk_cb(uv_handle_t* handle, void* arg) {
 
   // Sanity check
   // Most invalid values are large and refs are small, 0x1000000 is arbitrary.
-  assert(data && data->ref < 0x1000000);
+  if(!data || data->ref >= 0x1000000) return;
```

Comment:
The intent here is to ensure that the handles are created by Luv. Since the support for external (non-Luv-created) objects was not comprehensively considered in the original design of Luv, and there is no well-established mechanism to distinguish between external and internal objects, there are likely still issues present here.

Reply:
> there are likely still issues present here.

Should the solution instead tag all luv created handles, and check for the existence of that tag?  Simply making this a return seems fair to me

Reply:
Yeah, I understand. What's why I proposed tagging external handles with magic values to make luv exit early here. That'd be a NO-OP for luv tests and you can leave the assertion unchanged. If there are other issues  only relevant to this use case, I'll probably find them sooner or later (with automated testing). Or indeed, could tag ALL luv handles.

Reply:
I believe the intent here should be simply to not crash unexpectedly for the end-user over something they might not have control for in production, rather than guaranteeing they are getting luv handles, which I would argue is out of scope for this PR and can be done later on, perhaps by as suggested tagging all luv handles.

Reply:
It's hard to say; I don't have a safe and efficient solution at the moment. 

some options 

1.  tag, that with high performance, but not enough safe.
2. cache all data internal object pointer in a registry table, check it internal or external object,  that safe enough and and good performance.
3. other solution...

> > there are likely still issues present here.
> 
> Should the solution instead tag all luv created handles, and check for the existence of that tag? Simply making this a return seems fair to me